### PR TITLE
[DO NOT MERGE] Upgrade tensorflow

### PR DESCRIPTION
**Pull request by sihanwang41**
Signed-off-by: Sihan Wang <sihanwang41@gmail.com>

## Why are these changes needed?

tensorflow 2.6.2 has a restricted version control for the type_extensions which cause the issue to upgrade fast api (require different version of type_extensions)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/requirements/ml/requirements_dl.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # These requirements are used for the CI and CPU-only Docker images so we install CPU only versions of torch.
 # For GPU Docker images, you should install requirements_ml_docker.txt afterwards.
 
-tensorflow==2.6.2
+tensorflow==2.10.0
```
Comment (amogkam) [inline python/requirements/ml/requirements_dl.txt:4]: lets keep this as 2.6.3. It will introduce the least amount of changes.

**Review by amogkam — CHANGES_REQUESTED** [review]

File: python/requirements/ml/requirements_dl.txt (hunk 1 above)
Comment (amogkam) [reply python/requirements/ml/requirements_dl.txt:4]: only tensorflow 2.6.2 has the typing extensions restriction. 2.6.3 does not.

**Review by amogkam — APPROVED** [review]
Thanks, lgtm!

**Review by krfricke — APPROVED** [review]

**Comment by krfricke** [comment]
@sihanwang41 can you update the PR description?

**Comment by sihanwang41** [comment]
> @sihanwang41 can you update the PR description?

Updated!

**Comment by amogkam** [comment]
Closing this as it breaks rllib tests. Will have to revisit this later cc @ArturNiederfahrenhorst

**Pull request closed without merging**